File: aon-occam/src/main/java/com/esferalia/aon/occam/api/model/type/ElaborationStatus.java
package com.esferalia.aon.occam.api.model.type;


public enum ElaborationStatus {

    PENDING("Pendiente"),
    CLOSED("Cerrado"),
    IN_PROGRESS("En progreso"),
    FAIL("Fallido"),
    REOPEN("Reabierto"),
    ;
	
	private String name;
	
	private ElaborationStatus(String name) {
		this.name = name;
	}
	
    public String getName() {
		return name;
    }
    
	public byte value() {
		return (byte) this.ordinal();
	}
	
	public static ElaborationStatus safeValueOf( Byte i ) {
		if (i == null) return null;
		return safeValueOf( i.intValue() ); 
	}
	public static ElaborationStatus safeValueOf( Integer i ) {
		if (i == null) return null;
		if (i < 0 || i >= ElaborationStatus.values().length) return null;
		return ElaborationStatus.values()[i];
	}

}

File: aon-web-aio/src/main/webapp/models/elaboration/elaborationStatus.js
import { CONSTANT, MSG } from "../../environments/environments";

export const ElaborationStatus = {
    PENDING: {
        id: CONSTANT.PENDING,
        name: MSG.PENDING
    },
    CLOSED: {
        id: CONSTANT.CLOSED,
        name: MSG.CLOSED
    },
    IN_PROGRESS: {
        id: CONSTANT.IN_PROGRESS,
        name: MSG.IN_PROGRESS
    },
    FAIL: {
        id: CONSTANT.FAIL,
        name: MSG.FAILED
    },
    REOPEN: {
        id: CONSTANT.REOPEN,
        name: MSG.REOPENED
    }
}

export const elaborationStatuses = [
    {name: MSG.PENDING, value: CONSTANT.PENDING.toUpperCase()},
    {name: MSG.CLOSED, value: CONSTANT.CLOSED.toUpperCase()},
    {name: MSG.IN_PROGRESS, value: CONSTANT.IN_PROGRESS.toUpperCase()},
    {name: MSG.FAILED, value: CONSTANT.FAIL.toUpperCase()},
    {name: MSG.REOPENED, value: CONSTANT.REOPEN.toUpperCase()}

]